packages/core/core/prompt/data-content.ts

```diff
@@ -59,3 +59,17 @@ export function convertDataContentToUint8Array(
 
   throw new InvalidDataContentError({ content });
 }
+
+/**
+ * Converts a Uint8Array to a string of text.
+ *
+ * @param uint8Array - The Uint8Array to convert.
+ * @returns The converted string.
+ */
+export function convertUint8ArrayToText(uint8Array: Uint8Array): string {
+  try {
+    return new TextDecoder().decode(uint8Array);
+  } catch (error) {
+    throw new Error('Error decoding Uint8Array to text');
+  }
+}
```

Comment:
we should move into `uint8-utils.ts` eventually (in this PR it would create to much churn in the provider tho)